### PR TITLE
refactor(curriculum): authors workshop step 7 description and tests update

### LINKED ISSUE #57071: Refactor step 7 of authors workshop
## File to update 

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md

## Update description

```md
Now that you have the data you want, you can use it to populate the UI. 

But the data you are fetching will have a total of 26 authors. Instead of displaying all 26 authors at once, you should display 8 authors at a time.

Start by using the `let` keyword to create two variables named `startingIndex` and `endingIndex`, and assign them the numbers `0` and `8`, respectively. You will need to use `let` here because you will be reassigning these values later on.

Then, create an `authorDataArr` variable and assign it an empty array. 
```

## Updated tests

We only need regex tests to specifically check for `let`. So those tests can remain. The rest of these tests should use specific assert methods like `assert.equal`, `isNumber`, `isArray`, etc
https://www.chaijs.com/api/assert/

Also, these tests can be more granular

We should test for the following:

- You should use `let` to declare a variable named `startingIndex`.
- Your `startingIndex` variable should be a number.
- Your `startingIndex` variable should be set to `0`.
- You should use `let` to declare a variable named `endingIndex`.
- Your `endingIndex` variable should be a number.
- You should set your `endingIndex` variable to `8`.
- You should have a variable named `authorDataArr`.
- Your `authorDataArr` variable should be an empty array.

### PR DESCRIPTION
**Pull request by RGHANILOO**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #57071

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 1]
```diff
@@ -21,10 +23,16 @@ You should use `let` to declare a variable named `startingIndex`.
 assert.match(code, /let\s+startingIndex/)
 ```
 
-You should set your `startingIndex` variable to `0`.
+Your `startingIndex` variable should be a number.
 
 ```js
-assert.match(code, /let\s+startingIndex\s*=\s*0\s*;?/)
+assert.isNumber(startingIndex, 'startingIndex should be a number')
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:29]: we don't need comments for the second argument here

```suggestion
assert.isNumber(startingIndex);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 2]
```diff
@@ -21,10 +23,16 @@ You should use `let` to declare a variable named `startingIndex`.
...
+```
+
+Your `startingIndex` variable should be set to `0`.
+
+```js
+assert.strictEqual(startingIndex, 0, 'startingIndex should be set to 0')
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:35]:
```suggestion
assert.strictEqual(startingIndex, 0);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 3]
```diff
@@ -33,22 +41,29 @@ You should use `let` to declare a variable named `endingIndex`.
 assert.match(code, /let\s+endingIndex/)
 ```
 
+Your `endingIndex` variable should be a number.
+  
+  ```js
+assert.isNumber(endingIndex, 'endingIndex should be a number')
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:47]:
```suggestion
assert.isNumber(endingIndex);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 4]
```diff
@@ -33,22 +41,29 @@ You should use `let` to declare a variable named `endingIndex`.
...
+```
+
 You should set your `endingIndex` variable to `8`.
 
 ```js
-assert.match(code, /let\s+endingIndex\s*=\s*8\s*;?/)
+assert.strictEqual(endingIndex, 8, 'endingIndex should be set to 8')
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:53]:
```suggestion
assert.strictEqual(endingIndex, 8);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 5]
```diff
@@ -33,22 +41,29 @@ You should use `let` to declare a variable named `endingIndex`.
...
 ```
 
-You should use `let` to declare a variable named `authorDataArr`.
+You should have a variable named `authorDataArr`.
 
 ```js
-assert.match(code, /let\s+authorDataArr/)
+assert.isDefined(authorDataArr, 'authorDataArr should be defined');
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:59]:
```suggestion
assert.isDefined(authorDataArr);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 6]
```diff
@@ -33,22 +41,29 @@ You should use `let` to declare a variable named `endingIndex`.
...
 ```
 
-You should set your `authorDataArr` variable to an empty array (`[]`).
+Your `authorDataArr` variable should be an empty array.
 
 ```js
-assert.match(code, /let\s+authorDataArr\s*=\s*\[\s*\]\s*;?/)
+assert.isArray(authorDataArr, 'authorDataArr should be an array');
+assert.deepEqual(authorDataArr, [], 'authorDataArr should be an empty array');
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:66]: comments needs to be removed here like the other ones.
But also for the second assert, isEmpty would be better 
https://www.chaijs.com/api/assert/#method_isempty

Comment (RGHANILOO) [reply curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md]: thanks for the pointer.
latest commit resolves all your suggested changes

**Review by jdwilkin4 — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md [hunk 7]
```diff
@@ -7,48 +7,63 @@ dashedName: step-7
...
+assert.isDefined(authorDataArr);
 ```
 
-You should set your `authorDataArr` variable to an empty array (`[]`).
+Your `authorDataArr` variable should be an empty array.
 
 ```js
-assert.match(code, /let\s+authorDataArr\s*=\s*\[\s*\]\s*;?/)
+assert.isArray(authorDataArr);
+assert.isEmpty([]);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:66]: this last test, how is this useful? you are testing if an empty array is empty, it's never going to fail

Comment (jdwilkin4) [reply curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5462576784453146ec2.md:66]: oopps.. missed that on the last review

```suggestion
assert.isEmpty(authorDataArr);
```

**Review by Sembauke — APPROVED** [review]

**Pull request merged**